app/Models/Course.php

```diff
@@ -51,4 +53,20 @@ public function modules(): HasMany
     {
         return $this->hasMany(Module::class);
     }
+
+    public function isFinished(?User $user): bool
+    {
+        $isFinished = true;
+        foreach ($this->modules as $module) {
+            $moduleUser = $module->users->find($user->id)?->getRelationValue('pivot');
+            if ($moduleUser?->finished_content_at) {
+                continue;
+            } else {
+                $isFinished = false;
+                break;
+            }
```

Comment:
I took the first approach with additional logic to return false if there's no training modules. I couldn't figure out using the intermediate table queries, and I think it's because this is Course model, and $this would return a course instance.